test/06-batch.test.ts

```diff
@@ -0,0 +1,537 @@
+
+// TODO: figure out how to not persist the results across describes
+describe('06 batch tests', function () {
+	let pineServer: Awaited<ReturnType<typeof testInit>>;
+	before(async () => {
+		pineServer = await testInit({
+			configPath,
+			hooksPath,
+			deleteDb: true,
+		});
+		// setup faker so that test date uniqueness is set for all test cases
+		faker.seed();
+	});
+
+	after(async () => {
+		testDeInit(pineServer);
+	});
+
+	describe('Basic', () => {
+		it('check /ping route is OK', async () => {
+			await supertest(testLocalServer).get('/ping').expect(200, 'OK');
+		});
+	});
+
+	describe('test non-atomic batch requests', () => {
+		it('should create two students', async () => {
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id: '0',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100000,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'foo',
+							},
+						},
+						{
+							id: '1',
+							method: 'POST',
+							url: '/university/student',
+							body: {
+								matrix_number: 100001,
+								name: faker.name.firstName(),
+								last_name: faker.name.lastName(),
+								studies_at__campus: 'bar',
+							},
+						},
+					],
+				})
+				.expect(200);
+			const res = await supertest(testLocalServer)
+				.get('/university/student')
+				.expect(200);
+			expect(res.body)
+				.to.be.an('object')
+				.that.has.ownProperty('d')
+				.to.be.an('array')
+				.of.length(2);
+		});
+
+		it('successful request should have `responses` in its body', async () => {
+			const id = Math.random().toString();
+			const res = await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({
+					requests: [
+						{
+							id,
+							method: 'GET',
+							url: '/university/student',
+						},
+					],
+				})
+				.expect(200);
+			expect(res.body)
+				.to.be.an('object')
+				.that.has.ownProperty('responses')
+				.to.be.an('array')
+				.of.length(1);
+			expect(res.body.responses[0].body)
+				.to.be.an('object')
+				.that.has.ownProperty('d')
+				.to.be.an('array')
+				.of.length(2);
+			expect(res.body.responses[0].id).to.equal(id);
+		});
+
+		it('should fail if the body does not have a valid "requests" property', async () => {
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({})
+				.expect(
+					400,
+					'"Batch requests must include an array of requests in the body via the \\"requests\\" property"',
+				);
+			await supertest(testLocalServer)
+				.post('/university/$batch')
+				.send({ requests: 'test' })
+				.expect(
+					400,
+					'"Batch requests must include an array of requests in the body via the \\"requests\\" property"',
+				);
+		});
+
+		// TODO: Seems we have default `continue-on-error` = `false`, but the docs specify `true`. Do we want to continue like this?
```

Comment:
I think, as long as request cannot depend on each other, pine can execute all request in parallel. If one fails, pine just return back one failed request and other successful ones.

I don't see an implementation of dependsOn, so request are all independent and it's more a request bundling and not a chained execution of requests.
https://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_BatchRequest